tests/functional/s/simplifiable_condition.py

```diff
@@ -0,0 +1,41 @@
+"""Test that boolean conditions can be simplified"""
+
+# pylint: disable=pointless-statement
+
+
+def func(_):
+    """Pointless function"""
+
+
+CONSTANT = 100
+OTHER = 200
+
+# Simplifies any boolean expression that is coerced into a True/False value
+bool(CONSTANT or True)  # [simplifiable-condition]
+assert CONSTANT or False  # [simplifiable-condition]
+if CONSTANT and True:  # [simplifiable-condition]
+    pass
+while CONSTANT and False:  # [simplifiable-condition]
+    break
+1 if CONSTANT or True else 2  # [simplifiable-condition]
+
+# Simplifies recursively
+assert CONSTANT or (True and False)  # [simplifiable-condition]
+assert True and CONSTANT and OTHER  # [simplifiable-condition]
+assert True or CONSTANT or OTHER  # [simplifiable-condition]
+assert (CONSTANT or False) and (OTHER or True)  # [simplifiable-condition]
+assert (CONSTANT or True) or (CONSTANT or True)  # [simplifiable-condition]
+
+# Will try to infer the truthiness of an expression as long as it doesn't contain any variables
+assert [] or CONSTANT  # [simplifiable-condition]
+assert 3 + 4 or CONSTANT  # [simplifiable-condition]
+assert {} or CONSTANT  # [simplifiable-condition]
+
+assert True or True # [simplifiable-condition]
+assert False or False # [simplifiable-condition]
+assert True and True # [simplifiable-condition]
+assert False and False # [simplifiable-condition]
+
+# Expressions not in one of the above situations will not emit a message
+CONSTANT or True
```

Comment:
Shouldn't `bool(Unknown or True)` emit an error, as it can be simplified to `True`?

Reply:
Not quite, as we do not know what `Unknown` actually is at runtime.

Reply:
But in this case the value of Unknown shouldn't matter:
```
>>> Unknown = False
>>> bool(Unknown or True)
True
>>> Unknown = True
>>> bool(Unknown or True)
True
>>> Unknown = 123
>>> bool(Unknown or True)
True
```

Reply:
You are right, in the case of `bool()` it should not matter. Looking at the generated recommendation though, it seems that `assert Unknown or True` suggests to refactor the expression as `assert True`. In this particular case, shouldn't be `assert Unknown`, as hinted by your example as well? Only if `Unknown` is a falsey value, the second operand will be used, so the recommendation would result in the wrong code.

Reply:
From my understanding of the `assert` statement, it will raise an `AssertionError` if the condition is falsey, and will pass if it's truthy, thus the actual value of the boolean expression doesn't matter.
```
>>> CONSTANT = 0
>>> Unknown = 123
>>> assert Unknown or True
>>> Unknown = False
>>> assert Unknown or True
>>> assert False
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```
More broadly the checker in this PR only attempts to simplify boolean operations that are tested for truthiness and the actual value is not in use, which is:
- `while x:` 
- `if x:`
- `bool(x)`
- `assert x`
- `elif x:`
- `y if x else z`
- `y for y in z if x`
Now that I wrote out the list I realized I'm missing tests for the last 3 I listed so I'll add those in. I think `assert` falls into this category though, but please correct me if there's something I'm misunderstanding here!

Reply:
Regarding:

> assert Unknown or True suggests to refactor the expression as assert True

I think the language of the message could probably be more neutral we don't know if the problem is with the condition being plain wrong (ie : it should be `assert Unknown`) or really simplifiable (ie : it should be `assert True`). I think this kind of condition is more "suspicious" than "simplifiable".

Reply:
I'm not sure I understand what you mean by a condition being wrong vs simplifiable, could you explain that a bit more? I'm open to changing the wording of the message though 👍

Reply:
If there is a constant in your assert (ie `assert function() or True` the code is suspicious and probably wrong ? Why would you write an assert that is always True ? So simplifying to `assert True` is probably not the fix you're aiming for. Maybe it should be only `assert function()` or removing the assert entirely. I don't know if that makes sense.

Reply:
Ah okay, I see what you mean. Maybe this checker should be split into two separate messages, `simplifiable-boolean-condition`, where constants can be removed from the expression like `CONSTANT or False`, and `suspicious-boolean-condition` when the expression can be simplified to constant, like `CONSTANT and False`.

Reply:
That sounds really good. I don't know how easy it is to add this differentiation in this MR though. What do you think ? Thank you for rebasing by the way :)

Reply:
Can I move forward with splitting the checker, or should I wait for PCManticore's input on this?

Reply:
Yeah I think you can go on and do it. Looking at @PCManticore profile he seems to be quite busy with "not gihub thing" lately and he's probably spammed with other github issues anyway. With two positives review + [this comment](https://github.com/PyCQA/pylint/pull/3644#discussion_r438573062) I'm confident enough to merge myself.

Reply:
Okay, sounds good! Just one question about `suspicious-boolean-condition`, should this checker target single constants in conditionals, i.e. `if True:` or `bool(True)`?  Currently this checker would only emit a message if the expression can be reduced to a single constant, but not if the expression _is_ a single constant.

Reply:
Hmm, yes probably. If it's really hard to disentangle the code, we can probably create an issue to do it separately.

Reply:
`using-constant-test` should cover the case of single constants mentioned above